graphql/resolvers.go

```diff
@@ -3316,6 +3331,21 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 		if mainlineCommitVersion.Version != nil || mainlineCommitVersion.RolledUpVersions != nil {
 			mainlineCommits.Versions = append(mainlineCommits.Versions, &mainlineCommitVersion)
 		}
+		index++
+		// If we have exhausted all of our versions we should fetch some more.
+		if index == len(versions) {
```

Comment:
Right now we'll still be doing this query even if this is the version that gets us equal to limit, so we should either move this to the top of the loop or add an if to make sure we aren't doing that